scanpipe/tasks.py

```diff
@@ -76,3 +78,17 @@ def execute_pipeline_task(run_pk):
         project.clear_tmp_directory()
         if next_run := project.get_next_run():
             next_run.start()
+
+
+@job
+def background_delete_task(project):
+    # Check if the project is still marked for deletion
+    if not project.is_marked_for_deletion:
+        return
+
+    try:
+        project.delete()
+    except Exception as e:
+        info(f"Deletion failed: {str(e)}", project.pk)
+        project.update(is_marked_for_deletion=True)
```

Comment:
I'm wondering is this line is of any use, the `is_marked_for_deletion` must be already True to reach this line, no?